pkg/sql/catalog/descs/testdata/drop_and_alter_in_txn

```diff
@@ -0,0 +1,10 @@
+exec db=defaultdb
```

Comment:
what is the purpose of this test? In other words, how is it going to test the change that we no longer do cross-descriptor validation when "waitForOneVersion"?

Reply:
It's because we call `cleanupTempSystemTables()` [here](https://github.com/cockroachdb/cockroach/blob/003c0360de8b64319b5f0f127b99be91dbdca8a3/pkg/ccl/backupccl/restore_job.go#L2112) in a txn, where we run DDLs using an internal executor. Prior to this PR, if I migrate that to using `descs.TxnWithExector()` (which is the correct interface to run DDLs with internal executor), it will give a `descriptor not found` error.  This is because, in `descs.TxnWithExecutor()`, when running `WaitForOneVersion`, we weren’t waiting for the true descriptor id, rather we were waiting for something which referenced that id, and it was deleted.

See more discussion [here (internal link)](https://cockroachlabs.slack.com/archives/CARM0PTK4/p1661274424058389)